src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -15,32 +15,30 @@ namespace System.IO
     //
     public class BinaryWriter : IDisposable, IAsyncDisposable
     {
+        private const int MaxArrayPoolRentalSize = 1024 * 1024; // ArrayPool<T>.Shared allocates beyond this point
+
+        private delegate void WriteBufferToOutputDelegate(BinaryWriter sink, byte[] buffer, int offset, int count);
```

Comment:
It may be more efficient to just pass `bool` around to distinguish between these two cases.